### ISSUE TITLE
Having problems with GCM Core, it appears to not be installed

### ISSUE THREAD

**Issue by Rod-in-NM** [GitCredentialManager/git-credential-manager#519]
**Which version of GCM Core are you using?**

From a terminal, run `git-credential-manager-core --version` and paste the output.

> git-credential-manager-core : The term 'git-credential-manager-core' is not recognized as the name of a cmdlet,
function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the
path is correct and try again.
At line:1 char:1
> git-credential-manager-core --version
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     CategoryInfo          : ObjectNotFound: (git-credential-manager-core:String) [], CommandNotFoundException
>     FullyQualifiedErrorId : CommandNotFoundException

**Which Git host provider are you trying to connect to?**

* [ X] Azure DevOps
* [ ] Azure DevOps Server (TFS/on-prem)
* [ ] GitHub
* [ ] GitHub Enterprise
* [ ] Bitbucket
* [ ] Other - please describe

**Can you access the remote repository directly in the browser using the remote URL?**

Yes

From a terminal, run `git remote -v` to see your remote URL.

* [ ] Yes
* [ ] No, I get a permission error
* [ X] No, for a different reason - please describe

Since I cannot clone the remote repo, issuing the `git remote -v` command results in a "Fatal: not a git repository" error

---

**_[Azure DevOps only]_ What format is your remote URL?**

* [ ] Not applicable
* [ x] https://dev.azure.com/`{org}`/...
* [ ] https://`{org}`@dev.azure.com/`{org}`/...
* [ ] https://`{org}`.visualstudio.com/...

**_[Azure DevOps only]_ If the account picker shows more than one identity as you authenticate, check that you selected the same one that has access on the web.**

* [x ] Not applicable
* [ ] I only see one identity
* [ ] I checked each identity and none worked

It doesn't show me anything, other than ":" when asking me for the password. No identities are shown. Instead it gives me the error:

> git-credential-manager-core.exe: No such file or directory

---

**Expected behavior**

I am authenticated and my Git operation completes successfully.

**Actual behavior**

It gives me the error:

> git-credential-manager-core.exe: No such file or directory

then it asks me for a password.

**Logs**

Set the environment variables `GCM_TRACE=1` and `GIT_TRACE=1` and re-run your Git command. Review and redact any private information and attach the log.

I don't think I have GCM installed. On my Windows 10 Enterprise dev machine, there is no folder named ` Git Credential Manager Core` under my `Local/Programs` folder. I think that is the root cause of my problem. I am now at version 2.33.1.windows.1. My guess is that when I originally set this up, either GCM wasn't available or I didn't understand the importance of it, so I elected not to install it. If I'm correct, then how to I install GCM now? Would it be best if I uninstalled Git for Windows, then reinstall it afterwards?

**Comment by mjcheetham** [comment]
Please can you run the following and return with a copy of the output?

```text
# Command Prompt
dir "C:\Program Files\Git\mingw64\libexec\git-core"

# PowerShell
ls "C:\Program Files\Git\mingw64\libexec\git-core"
```

You should see `git-credential-manager-core.exe` listed. If you _do_ then please also run the following and report back with the output:

```text
git config --get-all --show-origin credential.helper
```

**Comment by Rod-in-NM** [comment]
Running the PowerShell command `ls "C:\Program Files\Git\mingw64\libexec\git-core"` I get the following result:

>     Directory: C:\Program Files\Git\mingw64\libexec\git-core
>
> Mode                 LastWriteTime         Length Name
> ----                 -------------         ------ ----
> d----          10/25/2021  9:47 AM                mergetools
-a---          10/13/2021 11:23 PM          44928 Atlassian.Bitbucket.dll
-a---          10/13/2021 11:23 PM          41888 Atlassian.Bitbucket.UI.exe
-a---          10/13/2021 11:23 PM            877 Atlassian.Bitbucket.UI.exe.config
-a---          10/13/2021 11:23 PM          11776 Atlassian.Bitbucket.UI.Shared.dll
-a---          10/13/2021 11:23 PM          54272 edit.dll
-a---          10/13/2021 11:06 PM          49294 git-add--interactive
-a---          10/13/2021 11:06 PM        3513856 git-add.exe
-a---          10/13/2021 11:06 PM        3513856 git-am.exe
-a---          10/13/2021 11:06 PM        3513856 git-annotate.exe
-a---          10/13/2021 11:06 PM        3513856 git-apply.exe
-a---          10/13/2021 11:06 PM        3513856 git-archive.exe
-a---          10/13/2021 11:06 PM           3865 git-bisect
-a---          10/13/2021 11:06 PM        3513856 git-bisect--helper.exe
-a---          10/13/2021 11:06 PM        3513856 git-blame.exe
-a---          10/13/2021 11:06 PM        3513856 git-branch.exe
-a---          10/13/2021 11:06 PM        3513856 git-bugreport.exe
-a---          10/13/2021 11:06 PM        3513856 git-bundle.exe
-a---          10/13/2021 11:06 PM        3513856 git-cat-file.exe
-a---          10/13/2021 11:06 PM        3513856 git-check-attr.exe
-a---          10/13/2021 11:06 PM        3513856 git-check-ignore.exe
-a---          10/13/2021 11:06 PM        3513856 git-check-mailmap.exe
-a---          10/13/2021 11:06 PM        3513856 git-check-ref-format.exe
-a---          10/13/2021 11:06 PM        3513856 git-checkout--worker.exe
-a---          10/13/2021 11:06 PM        3513856 git-checkout-index.exe
-a---          10/13/2021 11:06 PM        3513856 git-checkout.exe
-a---          10/13/2021 11:06 PM        3513856 git-cherry-pick.exe
-a---          10/13/2021 11:06 PM        3513856 git-cherry.exe
-a---          10/13/2021 11:06 PM            687 git-citool
-a---          10/13/2021 11:06 PM        3513856 git-clean.exe
-a---          10/13/2021 11:06 PM        3513856 git-clone.exe
-a---          10/13/2021 11:06 PM        3513856 git-column.exe
-a---          10/13/2021 11:06 PM        3513856 git-commit-graph.exe
-a---          10/13/2021 11:06 PM        3513856 git-commit-tree.exe
-a---          10/13/2021 11:06 PM        3513856 git-commit.exe
-a---          10/13/2021 11:06 PM        3513856 git-config.exe
-a---          10/13/2021 11:06 PM        3513856 git-count-objects.exe
-a---          10/13/2021 11:06 PM        3513856 git-credential-cache--daemon.exe
-a---          10/13/2021 11:06 PM        3513856 git-credential-cache.exe
-a---          10/13/2021 11:23 PM          74152 git-credential-manager-core.exe
-a---          10/13/2021 11:23 PM            877 git-credential-manager-core.exe.config
-a---          10/13/2021 11:06 PM        3513856 git-credential-store.exe
-a---          10/13/2021 11:06 PM          58928 git-credential-wincred.exe
-a---          10/13/2021 11:06 PM        3513856 git-credential.exe
-a---          10/13/2021 11:06 PM        1926160 git-daemon.exe
-a---          10/13/2021 11:06 PM        3513856 git-describe.exe
-a---          10/13/2021 11:06 PM        3513856 git-diff-files.exe
-a---          10/13/2021 11:06 PM        3513856 git-diff-index.exe
-a---          10/13/2021 11:06 PM        3513856 git-diff-tree.exe
-a---          10/13/2021 11:06 PM        3513856 git-diff.exe
-a---          10/13/2021 11:06 PM           2503 git-difftool--helper
-a---          10/13/2021 11:06 PM        3513856 git-difftool.exe
-a---          10/13/2021 11:06 PM        3513856 git-env--helper.exe
-a---          10/13/2021 11:06 PM        3513856 git-fast-export.exe
-a---          10/13/2021 11:06 PM        3513856 git-fast-import.exe
-a---          10/13/2021 11:06 PM        3513856 git-fetch-pack.exe
-a---          10/13/2021 11:06 PM        3513856 git-fetch.exe
-a---          10/13/2021 11:06 PM          15862 git-filter-branch
-a---          10/13/2021 11:06 PM        3513856 git-fmt-merge-msg.exe
-a---          10/13/2021 11:06 PM        3513856 git-for-each-ref.exe
-a---          10/13/2021 11:06 PM        3513856 git-for-each-repo.exe
-a---          10/13/2021 11:06 PM        3513856 git-format-patch.exe
-a---          10/13/2021 11:06 PM        3513856 git-fsck-objects.exe
-a---          10/13/2021 11:06 PM        3513856 git-fsck.exe
-a---          10/13/2021 11:06 PM        3513856 git-fsmonitor--daemon.exe
-a---          10/13/2021 11:06 PM        3513856 git-gc.exe
-a---          10/13/2021 11:06 PM        3513856 git-get-tar-commit-id.exe
-a---          10/13/2021 11:06 PM        3513856 git-grep.exe
-a---          10/13/2021 11:06 PM            687 git-gui
-a---          10/13/2021 11:06 PM           1840 git-gui--askpass
-a---          10/13/2021 11:06 PM           1605 git-gui--askyesno
-a---          10/13/2021 11:06 PM         106619 git-gui.tcl
-a---          10/13/2021 11:06 PM        3513856 git-hash-object.exe
-a---          10/13/2021 11:06 PM        3513856 git-help.exe
-a---          10/13/2021 11:06 PM        1923088 git-http-backend.exe
-a---          10/13/2021 11:06 PM        2105872 git-http-fetch.exe
-a---          10/13/2021 11:06 PM        2117136 git-http-push.exe
-a---          10/13/2021 11:06 PM        2113552 git-imap-send.exe
-a---          10/13/2021 11:06 PM        3513856 git-index-pack.exe
-a---          10/13/2021 11:06 PM        3513856 git-init-db.exe
-a---          10/13/2021 11:06 PM        3513856 git-init.exe
-a---          10/13/2021 11:06 PM          22370 git-instaweb
-a---          10/13/2021 11:06 PM        3513856 git-interpret-trailers.exe
-a---          10/13/2021 11:06 PM        3513856 git-log.exe
-a---          10/13/2021 11:06 PM        3513856 git-ls-files.exe
-a---          10/13/2021 11:06 PM        3513856 git-ls-remote.exe
-a---          10/13/2021 11:06 PM        3513856 git-ls-tree.exe
-a---          10/13/2021 11:06 PM        3513856 git-mailinfo.exe
-a---          10/13/2021 11:06 PM        3513856 git-mailsplit.exe
-a---          10/13/2021 11:06 PM        3513856 git-maintenance.exe
-a---          10/13/2021 11:06 PM        3513856 git-merge-base.exe
-a---          10/13/2021 11:06 PM        3513856 git-merge-file.exe
-a---          10/13/2021 11:06 PM        3513856 git-merge-index.exe
-a---          10/13/2021 11:06 PM           2477 git-merge-octopus
-a---          10/13/2021 11:06 PM           3695 git-merge-one-file
-a---          10/13/2021 11:06 PM        3513856 git-merge-ours.exe
-a---          10/13/2021 11:06 PM        3513856 git-merge-recursive.exe
-a---          10/13/2021 11:06 PM            944 git-merge-resolve
-a---          10/13/2021 11:06 PM        3513856 git-merge-subtree.exe
-a---          10/13/2021 11:06 PM        3513856 git-merge-tree.exe
-a---          10/13/2021 11:06 PM        3513856 git-merge.exe
-a---          10/13/2021 11:06 PM          11564 git-mergetool
-a---          10/13/2021 11:06 PM           9772 git-mergetool--lib
-a---          10/13/2021 11:06 PM        3513856 git-mktag.exe
-a---          10/13/2021 11:06 PM        3513856 git-mktree.exe
-a---          10/13/2021 11:06 PM        3513856 git-multi-pack-index.exe
-a---          10/13/2021 11:06 PM        3513856 git-mv.exe
-a---          10/13/2021 11:06 PM        3513856 git-name-rev.exe
-a---          10/13/2021 11:06 PM        3513856 git-notes.exe
-a---          10/13/2021 11:06 PM         169268 git-p4
-a---          10/13/2021 11:06 PM        3513856 git-pack-objects.exe
-a---          10/13/2021 11:06 PM        3513856 git-pack-redundant.exe
-a---          10/13/2021 11:06 PM        3513856 git-pack-refs.exe
-a---          10/13/2021 11:06 PM        3513856 git-patch-id.exe
-a---          10/13/2021 11:06 PM        3513856 git-prune-packed.exe
-a---          10/13/2021 11:06 PM        3513856 git-prune.exe
-a---          10/13/2021 11:06 PM        3513856 git-pull.exe
-a---          10/13/2021 11:06 PM        3513856 git-push.exe
-a---          10/13/2021 11:06 PM           3693 git-quiltimport
-a---          10/13/2021 11:06 PM        3513856 git-range-diff.exe
-a---          10/13/2021 11:06 PM        3513856 git-read-tree.exe
-a---          10/13/2021 11:06 PM          29253 git-rebase--preserve-merges
-a---          10/13/2021 11:06 PM        3513856 git-rebase.exe
-a---          10/13/2021 11:06 PM        3513856 git-receive-pack.exe
-a---          10/13/2021 11:06 PM        3513856 git-reflog.exe
-a---          10/13/2021 11:06 PM        3513856 git-remote-ext.exe
-a---          10/13/2021 11:06 PM        3513856 git-remote-fd.exe
-a---          10/13/2021 11:06 PM        2121744 git-remote-ftp.exe
-a---          10/13/2021 11:06 PM        2121744 git-remote-ftps.exe
-a---          10/13/2021 11:06 PM        2121744 git-remote-http.exe
-a---          10/13/2021 11:06 PM        2121744 git-remote-https.exe
-a---          10/13/2021 11:06 PM        3513856 git-remote.exe
-a---          10/13/2021 11:06 PM        3513856 git-repack.exe
-a---          10/13/2021 11:06 PM        3513856 git-replace.exe
-a---          10/13/2021 11:06 PM           4130 git-request-pull
-a---          10/13/2021 11:06 PM        3513856 git-rerere.exe
-a---          10/13/2021 11:06 PM        3513856 git-reset.exe
-a---          10/13/2021 11:06 PM        3513856 git-restore.exe
-a---          10/13/2021 11:06 PM        3513856 git-rev-list.exe
-a---          10/13/2021 11:06 PM        3513856 git-rev-parse.exe
-a---          10/13/2021 11:06 PM        3513856 git-revert.exe
-a---          10/13/2021 11:06 PM        3513856 git-rm.exe
-a---          10/13/2021 11:06 PM          62795 git-send-email
-a---          10/13/2021 11:06 PM        3513856 git-send-pack.exe
-a---          10/13/2021 11:06 PM           2031 git-sh-i18n
-a---          10/13/2021 11:06 PM        1910800 git-sh-i18n--envsubst.exe
-a---          10/13/2021 11:06 PM           9726 git-sh-setup
-a---          10/13/2021 11:06 PM        3513856 git-shortlog.exe
-a---          10/13/2021 11:06 PM        3513856 git-show-branch.exe
-a---          10/13/2021 11:06 PM        3513856 git-show-index.exe
-a---          10/13/2021 11:06 PM        3513856 git-show-ref.exe
-a---          10/13/2021 11:06 PM        3513856 git-show.exe
-a---          10/13/2021 11:06 PM        3513856 git-sparse-checkout.exe
-a---          10/13/2021 11:06 PM        3513856 git-stage.exe
-a---          10/13/2021 11:06 PM        3513856 git-stash.exe
-a---          10/13/2021 11:06 PM        3513856 git-status.exe
-a---          10/13/2021 11:06 PM        3513856 git-stripspace.exe
-a---          10/13/2021 11:06 PM          19256 git-submodule
-a---          10/13/2021 11:06 PM        3513856 git-submodule--helper.exe
-a---          10/13/2021 11:06 PM          23570 git-subtree
-a---          10/13/2021 11:06 PM          65844 git-svn
-a---          10/13/2021 11:06 PM        3513856 git-switch.exe
-a---          10/13/2021 11:06 PM        3513856 git-symbolic-ref.exe
-a---          10/13/2021 11:06 PM        3513856 git-tag.exe
-a---          10/13/2021 11:06 PM        3513856 git-unpack-file.exe
-a---          10/13/2021 11:06 PM        3513856 git-unpack-objects.exe
-a---          10/13/2021 11:23 PM            218 git-update
-a---          10/13/2021 11:06 PM        3513856 git-update-index.exe
-a---          10/13/2021 11:06 PM        3513856 git-update-ref.exe
-a---          10/13/2021 11:06 PM        3513856 git-update-server-info.exe
-a---          10/13/2021 11:06 PM        3513856 git-upload-archive.exe
-a---          10/13/2021 11:06 PM        3513856 git-upload-pack.exe
-a---          10/13/2021 11:06 PM        3513856 git-var.exe
-a---          10/13/2021 11:06 PM        3513856 git-verify-commit.exe
-a---          10/13/2021 11:06 PM        3513856 git-verify-pack.exe
-a---          10/13/2021 11:06 PM        3513856 git-verify-tag.exe
-a---          10/13/2021 11:06 PM           4401 git-web--browse
-a---          10/13/2021 11:06 PM        3513856 git-whatchanged.exe
-a---          10/13/2021 11:06 PM        3513856 git-worktree.exe
-a---          10/13/2021 11:06 PM        3513856 git-write-tree.exe
-a---          10/13/2021 11:06 PM        3513856 git.exe
-a---          10/13/2021 11:23 PM          78216 GitHub.dll
-a---          10/13/2021 11:23 PM          44928 GitHub.UI.exe
-a---          10/13/2021 11:23 PM            877 GitHub.UI.exe.config
-a---          10/13/2021 11:23 PM          17920 GitHub.UI.Shared.dll
-a---          10/13/2021 11:06 PM          32784 headless-git.exe
-a---          10/13/2021 11:23 PM         142885 libbrotlicommon.dll
-a---          10/13/2021 11:23 PM          52364 libbrotlidec.dll
-a---          10/13/2021 11:23 PM          99146 libbz2-1.dll
-a---          10/13/2021 11:23 PM         122833 libcares-4.dll
-a---          10/13/2021 11:23 PM        2783388 libcrypto-1_1-x64.dll
-a---          10/13/2021 11:23 PM         675484 libcurl-4.dll
-a---          10/13/2021 11:23 PM         202568 libexpat-1.dll
-a---          10/13/2021 11:23 PM          82099 libgcc_s_seh-1.dll
-a---          10/13/2021 11:23 PM         509898 libgmp-10.dll
-a---          10/13/2021 11:23 PM         275300 libhogweed-6.dll
-a---          10/13/2021 11:23 PM        1058528 libiconv-2.dll
-a---          10/13/2021 11:23 PM         170241 libidn2-0.dll
-a---          10/13/2021 11:23 PM         133659 libintl-8.dll
-a---          10/13/2021 11:23 PM          78899 libjansson-4.dll
-a---          10/13/2021 11:23 PM         529915 libjemalloc.dll
-a---          10/13/2021 11:23 PM         153747 liblzma-5.dll
-a---          10/13/2021 11:23 PM         289910 libnettle-8.dll
-a---          10/13/2021 11:23 PM         177752 libnghttp2-14.dll
-a---          10/13/2021 11:23 PM         281695 libpcre-1.dll
-a---          10/13/2021 11:23 PM         618344 libpcre2-8-0.dll
-a---          10/13/2021 11:23 PM          41752 libpcreposix-0.dll
-a---          10/13/2021 11:23 PM         270123 libssh2-1.dll
-a---          10/13/2021 11:23 PM         562844 libssl-1_1-x64.dll
-a---          10/13/2021 11:23 PM          43431 libssp-0.dll
-a---          10/13/2021 11:23 PM        1746067 libstdc++-6.dll
-a---          10/13/2021 11:23 PM          92313 libtre-5.dll
-a---          10/13/2021 11:23 PM        1764460 libunistring-2.dll
-a---          10/13/2021 11:23 PM          59645 libwinpthread-1.dll
-a---          10/13/2021 11:23 PM        1358585 libxml2-2.dll
-a---          10/13/2021 11:23 PM         999818 libzstd.dll
-a---          10/13/2021 11:23 PM          56200 Microsoft.AzureRepos.dll
-a---          10/13/2021 11:23 PM         248712 Microsoft.Git.CredentialManager.dll
-a---          10/13/2021 11:23 PM          21888 Microsoft.Git.CredentialManager.UI.dll
-a---          10/13/2021 11:23 PM          26112 Microsoft.Git.CredentialManager.UI.Windows.dll
-a---          10/13/2021 11:23 PM         109016 Microsoft.Identity.Client.Desktop.dll
-a---          10/13/2021 11:23 PM        1424312 Microsoft.Identity.Client.dll
-a---          10/13/2021 11:23 PM          61944 Microsoft.Identity.Client.Extensions.Msal.dll
-a---          10/13/2021 11:23 PM         143264 Microsoft.Web.WebView2.Core.dll
-a---          10/13/2021 11:23 PM          28568 Microsoft.Web.WebView2.WinForms.dll
-a---          10/13/2021 11:23 PM          31128 Microsoft.Web.WebView2.Wpf.dll
-a---          10/13/2021 11:23 PM         700336 Newtonsoft.Json.dll
-a---          10/13/2021 11:23 PM          20856 System.Buffers.dll
-a---          10/13/2021 11:23 PM         197512 System.CommandLine.dll
-a---          10/13/2021 11:23 PM         141184 System.Memory.dll
-a---          10/13/2021 11:23 PM         115856 System.Numerics.Vectors.dll
-a---          10/13/2021 11:23 PM          16768 System.Runtime.CompilerServices.Unsafe.dll
-a---          10/13/2021 11:23 PM        1679679 tcl86.dll
-a---          10/13/2021 11:23 PM        1505405 tk86.dll
-a---          10/13/2021 11:23 PM         109984 WebView2Loader.dll
-a---          10/13/2021 11:23 PM         116428 zlib1.dll

Then when I ran the command `git config --get-all --show-origin credential.helper` I get the following:

> file:C:/Program Files/Git/etc/gitconfig manager-core
file:I://.gitconfig     manager
file:I://.gitconfig
file:I://.gitconfig     C:/Users/rod.falanga/AppData/Local/Programs/Git\ Credential\ Manager\ Core/git-credential-manager-core.exe

**Comment by mjcheetham** [comment]
Ahh right. Can I ask you to check if you have the standalone GCM Core (User) installed?

Windows Settings > Apps > Apps & features > search for "Git Credential Manager" and uninstall any entries that you find.

Then run the following commands to fix up your Git configuration:

```powershell
git config --global --unset-all credential.helper
```

Can you then update your Git for Windows to the latest, and make sure to select "Git Credential Manager Core" from the installation screen about selecting a credential helper.

To check you're using GCM Core you should see `manager-core` as the output from `git config credential.helper`.

**Comment by Rod-in-NM** [comment]
I've followed your instructions, with one exception. When I got to the point of updating Git for Windows, it never asked me to select "Git Credential Manager Core". There was no installation screen asking me to select a credential helper.

However, when it was finished, I issued the command `git config credential.helper`. This is the response I got:

> C:/Users/rod.falanga/AppData/Local/Programs/Git\ Credential\ Manager\ Core/git-credential-manager-core.exe
